### DISCUSSION TITLE
Support for testing components as Interactions

### DISCUSSION THREAD

**Discussion by IvaanTorres** [histoire-dev/histoire#229] [Ideas]
It would be awesome to implement some kind of testing for those components as StoryBook does, using Interactions.
Like so, we could see a visual result of how a component or a whole view is used by the final user. 
It'd be a sort of E2E testing way to show the real functionality and interaction of the components stored in Histoire with the difference that the components can be manually edited and set up to change the logic of the interaction.

**Comment by alvarosabu** [comment]
Hi @IvaanTorres do you mean a Cypress Component testing integration?

**Reply by IvaanTorres** [reply]
Yes, kind of.
However, using this testing tool inside Histoire would give more isolated documentation to the components and the UI itself.
Basically, you would have the component and the option to run the functional test. At the end, we have isolated everything related to the component.
I was thinking that, as it would be much effort to create something like that from scratch, it wouldn't be possible to give access to Cypress to run the component test from Histoire, manually? I think it could give more readability to the components since you run it from Histoire, inside the component itself.
It's just a supposition.
